### DISCUSSION TITLE
Unable to call my custom filter

### DISCUSSION THREAD

**Discussion by zeshan77** [spatie/laravel-query-builder#672] [Q&A]
Hi,

I'm using custom filter option of the package using the following code:

```php
QueryBuilder::for($this->query())
     ->allowedFilters([
            AllowedFilter::custom('role', new UserRoleFilter),
         ])
      ->....
```

`UserRoleFilter` class:
```php
use Illuminate\Database\Eloquent\Builder;
use Spatie\QueryBuilder\Filters\Filter;

class UserRoleFilter implements Filter
{

    public function __invoke(Builder $query, $value, string $property): void
    {
        dd($value, $property);
    }

}

```

Here's my http request:
`http://url.com?filter[role]=admin&sort=status&page=1&perPage=50`

The above codebase doesn't call the `UserRoleFilter __invoke()` for some reason. I'm wondering to know if I'm doing something wrong or I'm missing something.

Thanks

**Comment by david34corbalan** [comment]
I am wondering if the filter with only 1 value worked for you, because I have a filter from the model user and it is not working for me with more than 1 parameter. The result is wrong every time I try it.

https://github.com/spatie/laravel-query-builder/discussions/671
